### PR TITLE
Not using delegate_to ESXi host task and fix check EFI issue

### LINKED ISSUE #146: Consider using vsphere_copy and/or vsphere_file modules for ESXi datastore file operations
##### SUMMARY

Try to use "vsphere_copy" module instead of copy in "common/transfer_file_remote.yml" to do file transferring to ESXi datastore if it works to avoid operations on ESXi host through SSH directly.

##### ISSUE TYPE
- Feature Idea

##### ENVIRONMENT / OS

ESXi

### LINKED ISSUE #179: [New Feature] not use delegate_to ESXi host to find keyword in vmware.log
##### SUMMARY

Find other ways to search keyword in VM vmware.log instead of execute command through `delegate_to` ESXi host.
```
./common/vm_search_vmware_log.yml:35:      delegate_to: "{{ esxi_hostname }}"
./common/vm_wait_gosc_completed.yml:17:  delegate_to: "{{ esxi_hostname }}"
./windows/utils/win_empty_vmware_log.yml:12:  delegate_to: "{{ esxi_hostname }}"
./windows/utils/win_check_winbsod.yml:12:  delegate_to: "{{ esxi_hostname }}"
./windows/guest_customization/check_guest_reset_time.yml:17:      delegate_to: "{{ esxi_hostname }}"
```

##### ISSUE TYPE
- Feature Idea

##### ENVIRONMENT / OS

All

### LINKED ISSUE #244: [New Feature] Make check_efi_firmware more flexible when check keywork in vmware.log
##### SUMMARY

Log info in vmware.log about EFI will be changed, so we need to modify this test case to be more flexible to adaptable to different releases.
Now it's 
```
2021-11-01T02:49:30.791Z| vmx| I125: The EFI ROM is 64-bit.
```

##### ISSUE TYPE
- Feature Idea

##### ENVIRONMENT / OS

All

### PR DESCRIPTION
**Pull request by Tomorrow9**
Fixes #146
Fixes #179 
Fixes #244 
1. Change `vm_wait_log_msg.yml` to accept regular expression.
2. Change `transfer_file_remote.yml` to `esxi_upload_datastore_file.yml` in Windows test cases.

### CODE REVIEW

File: linux/check_efi_firmware/check_efi_firmware.yml [hunk 1]
```diff
@@ -31,7 +31,7 @@
         - include_tasks: ../../common/vm_wait_log_msg.yml
           vars:
             vm_wait_log_name: "vmware.log"
-            vm_wait_log_msg: "The EFI ROM is {{ bitness }}"
+            vm_wait_log_msg: "The EFI ROM is .* {{ bitness }}"
```
Comment (keirazhang) [inline linux/check_efi_firmware/check_efi_firmware.yml:34]: It should have no space following .*, otherwise it won't match The EFI ROM is 64-bit.

Comment (keirazhang) [reply linux/check_efi_firmware/check_efi_firmware.yml:34]: By the way, can we add a task to fetch vmware.log in rescue block if check_efi_firmware failed?

Comment (Tomorrow9) [reply linux/check_efi_firmware/check_efi_firmware.yml:34]: Test this it works.
```
TASK [debug] *****************************************************************************************************************************
ok: [localhost] => {
    "msg": [
        "Found 'The EFI ROM .* 64-bit' message in VM log vmware.log: Success",
        "Found logs list: ['The EFI ROM is 64-bit']"
    ]
}
```

Comment (Tomorrow9) [reply linux/check_efi_firmware/check_efi_firmware.yml:34]: > By the way, can we add a task to fetch vmware.log in rescue block if check_efi_firmware failed?

This is another task or issue we can address for all the cases or some required cases.

**Review by keirazhang — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**